### PR TITLE
[DN] Refactoring designer news login and moving DN files to DN packages

### PR DESCRIPTION
**Pull request by florina-muntenescu**
Creating a feature package in `activities` and `app` modules. The goal is to have all Designer News related items in those packages, decrease the number of dependencies and then move all files to a designernews feature module. 

Extracting part of the login logic from `DesignerNewsPrefs` to corresponding login repositories and local data source. 

Making the first step towards having a way of providing dependencies to classes, by creating an `Injection.kt` file, that provides several classes for Designer News.

### CODE REVIEW

File: app/src/main/java/io/plaidapp/designernews/login/data/DesignerNewsLoginRepository.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ */
+
+package io.plaidapp.designernews.login.data
+
+import android.text.TextUtils
+import io.plaidapp.designernews.data.api.model.User
+
+class DesignerNewsLoginRepository(private val storage: DesignerNewsLoginLocalStorage) {
+    var accessToken: String? = null
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/designernews/login/data/DesignerNewsLoginRepository.kt]: Nit: New line between class and property declaration.

File: app/src/main/java/io/plaidapp/designernews/Injection.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+fun provideDesignerNewsRepository(context: Context): DesignerNewsRepository {
+    return provideDesignerNewsRepository(DesignerNewsPrefs.get(context).api)
+}
+
+fun provideDesignerNewsRepository(service: DesignerNewsService): DesignerNewsRepository {
+    return DesignerNewsRepository.getInstance(service)
+}
+
+private fun getHttpLoggingInterceptor(): HttpLoggingInterceptor {
+    val debugLevel = if (BuildConfig.DEBUG)
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/designernews/Injection.kt]: Nit: Braces missing

File: app/src/main/java/io/plaidapp/designernews/Injection.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        HttpLoggingInterceptor.Level.BASIC
+    else
+        HttpLoggingInterceptor.Level.NONE
+    val loggingInterceptor = HttpLoggingInterceptor()
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/designernews/Injection.kt]: can be inlined and shortened with `.also {...}`

File: app/src/main/java/io/plaidapp/data/DataManager.java [hunk 4]
```diff
@@ -53,11 +53,9 @@
     private Map<String, Call> inflight;
 
     public DataManager(Context context,
-                       FilterAdapter filterAdapter) {
+            FilterAdapter filterAdapter) {
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/data/DataManager.java]: Nit: join lines instead

**Comment by florina-muntenescu** [comment]
ready for another review

**Pull request merged**